### ISSUE TITLE
Embedded YouTube video probably not GDPR compliant

### ISSUE THREAD

**Issue by jmenglund** [quarto-dev/quarto-cli#8212]
### Bug description

I embedded a YouTube video on a website using the following code:

````
{{< video https://www.youtube.com/embed/bRFHyI6hnNk >}}
````

I also set the cookie-consent policy in `_quarto.yml` to the most restrictive level:

````
website:
    cookie-consent:
        type: express
````

What I then discovered was that it was possible to interact with the YouTube video even if I had not consented to non-essential cookies. As I understand the situation, YouTube will process my personal data as soon as I interact with the embedded video (see for example https://axbom.com/embed-youtube-videos-without-cookies/). If the cookie-consent type is set to "express" I don't think that this should be possible, unless I have explicitly given the consent.

### Steps to reproduce

See the bug description above.

### Expected behavior

If the cookie consent-type for the website is set to "express", the user should not be able to interact directly with the embedded YouTube video, unless first accepting YouTube's terms and conditions.

### Actual behavior

A user can interact with the embedded YouTube video even if all cookie types except "strictly necessary cookies" are set to "Inactive".

### Your environment

- MacOS Sonoma 14.2
- Safari 17.2.1

### Quarto check output

```
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.3.450
      Path: /Applications/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.12 (Conda)
      Path: /opt/homebrew/Caskroom/miniforge/base/bin/python
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with conda install jupyter

[✓] Checking R installation...........OK
      Version: 4.3.2
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library
      knitr: 1.45
      rmarkdown: 2.25

[✓] Checking Knitr engine render......OK

```

**Comment by cscheid** [comment]
> If the cookie consent-type for the website is set to "express", the user should not be able to interact directly with the embedded YouTube video, unless first accepting YouTube's terms and conditions.

I don't know how Quarto can reason about these kinds of situations when the video shortcode can take an arbitrary URL where different providers will have their own downstream policies.

**Comment by allefeld** [comment]
I agree that Quarto devs can't possibly cover concerns about content embedded by users.

However, a similar concern may apply to fonts loaded from Google Fonts by Quarto themes. There are a number of threads about this on the Google Fonts GitHub repository, and I have to say that after reading through many of them I'm still not clear about the topic. What is strange is that at some point a representative of Google Fonts wrote that "The Google Fonts Web API does not log or store IP addresses", <https://github.com/google/fonts/issues/5463#issuecomment-1297422332>, but the Google Fonts FAQ entry they linked to does no longer exist. So it's unclear if this is really guaranteed by Google.

That means that a Quarto user in the EU who uses a theme which embeds a font from Google Fonts (e.g. Cosmo using Source Sans Pro) and pushes the result to a public website may be liable. Possibly. Maybe.

Note that this cannot be solved by a Cookie notice; apparently the problem is that the visitor's IP gets transferred to Google. Either the font has to be hosted locally, or there needs to be a landing page which informs and asks for consent, without using that font.

Just FYI, I'm not sure how you can address this if at all.

**Comment by cscheid** [comment]
You can use fonts from fonts.bunny.net if you so choose.

**Comment by allefeld** [comment]
How about putting that information on [HTML Theming](https://quarto.org/docs/output-formats/html-themes.html)?
